src/overlays/actors/ovl_Bg_Icicle/z_bg_icicle.c

```diff
@@ -21,26 +52,199 @@ const ActorInit Bg_Icicle_InitVars = {
     (ActorFunc)BgIcicle_Update,
     (ActorFunc)BgIcicle_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Icicle_0x809C9A60/BgIcicle_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneScale, 1500, ICHAIN_CONTINUE),
+    ICHAIN_F32(gravity, -3, ICHAIN_CONTINUE),
+    ICHAIN_F32(minVelocityY, -30, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx D_060000D0[];
+extern CollisionHeader D_06000294;
+
+void BgIcicle_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIcicle* this = THIS;
+
+    s32 paramsHigh;
+    s32 paramsMid;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000294);
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    Collider_UpdateCylinder(thisx, &this->collider);
+
+    paramsHigh = (thisx->params >> 8) & 0xFF;
+    paramsMid = (thisx->params >> 2) & 0x3F;
+    this->unk_161 = (thisx->params >> 8) & 0xFF;
+    thisx->params &= 3;
+
+    if (thisx->params == ICICLE_STALAGMITE_RANDOM_DROP || thisx->params == ICICLE_STALAGMITE_FIXED_DROP) {
+        this->unk_160 = ((thisx->params == ICICLE_STALAGMITE_RANDOM_DROP) ? paramsHigh : paramsMid);
+        this->actionFunc = BgIcicle_DoNothing;
+    } else {
+        this->dyna.actor.shape.rot.x = -0x8000;
+        this->dyna.actor.shape.yOffset = 1200.0f;
+        this->actionFunc = BgIcicle_Wait;
+    }
+}
+
+void BgIcicle_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIcicle* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgIcicle_Break(BgIcicle* this, GlobalContext* globalCtx, f32 arg2) {
+    static Vec3f accel = { 0.0f, -1.0f, 0.0f };
+    static Color_RGBA8 primColor = { 170, 255, 255, 255 };
+    static Color_RGBA8 envColor = { 0, 50, 100, 255 };
+    Vec3f velocity;
+    Vec3f pos;
+    s32 j;
+    s32 i;
+
+    func_800F0568(globalCtx, &this->dyna.actor.world.pos, 30, 0x28CB);
```

Comment:
```suggestion
    func_800F0568(globalCtx, &this->dyna.actor.world.pos, 30, NA_SE_EV_ICE_BROKEN);
```
These can be filled in with the most recent commit now :)